src/snapred/backend/recipe/algorithm/SmoothDataExcludingPeaksAlgo.py

```diff
@@ -0,0 +1,111 @@
+# Logic notes:
+"""
+    load nexus
+    num spe = num groups (instrument state) --> consitant with workspace
+    state & crystal info --> consitant with workspace
+    call diffraction spectrum peak predictor --> to get peaks (num groups)
+    call diffraction spectrum weight calc (send one spectra only)
+    extract x and y data from a workspace (numpy arrray)
+    implement csaps
+    create new workspace with csaps data
+"""
+import json
+
+from mantid.api import (
+    AlgorithmFactory,
+    PythonAlgorithm,
+    WorkspaceGroup,
+    mtd,
+)
+from mantid.kernel import Direction
+from scipy.interpolate import splev, splrep
+
+from snapred.backend.dao.SmoothDataExcludingPeaksIngredients import SmoothDataExcludingPeaksIngredients
+from snapred.backend.recipe.algorithm.DiffractionSpectrumWeightCalculator import DiffractionSpectrumWeightCalculator
+from snapred.backend.recipe.algorithm.MantidSnapper import MantidSnapper
+
+name = "SmoothDataExcludingPeaks"
+
+
+class SmoothDataExcludingPeaks(PythonAlgorithm):
+    def PyInit(self):
+        # declare properties
+        self.declareProperty("InputWorkspace", defaultValue="", direction=Direction.Input)
+        self.declareProperty("SmoothDataExcludingPeaksIngredients", defaultValue="", direction=Direction.Input)
+        self.declareProperty("OutputWorkspace", defaultValue="", direction=Direction.Output)
+        self.setRethrows(True)
+        self.mantidSnapper = MantidSnapper(self, name)
+
+    def PyExec(self):
+        self.log().notice("Removing peaks and smoothing data")
+
+        # load ingredients
+        smoothdataIngredients = SmoothDataExcludingPeaksIngredients(
+            **json.loads(self.getProperty("SmoothDataExcludingPeaksIngredients").value)
+        )
+
+        # load workspace
+        input_ws = self.getProperty("InputWorkspace").value
+        self.mantidSnapper.mtd[input_ws]
+
+        # load instrument state
+        instrumentState = smoothdataIngredients.InstrumentState
+
+        # load crystal info
+        crystalInfo = smoothdataIngredients.CrystalInfo
+
+        ws = self.mantidSnapper.mtd[input_ws]
+
+        # call the diffraction spectrum weight calculator
+        weight_ws_name = "weight_ws"
+        weightCalAlgo = DiffractionSpectrumWeightCalculator()
+        weightCalAlgo.initialize()
+        weightCalAlgo.setProperty("InputWorkspace", input_ws)
+        weightCalAlgo.setProperty("InstrumentState", instrumentState.json())
+        weightCalAlgo.setProperty("CrystalInfo", crystalInfo.json())
+        weightCalAlgo.setProperty("WeightWorkspace", weight_ws_name)
+        weightCalAlgo.execute()
```

Comment:
Similarly, this can be
```
self.mantidSnapper.DiffractionSpectrumWeightCalculator(
        InputWorkspace = input_ws,
        InstrumentState = smoothdataIngredients.instrumentState,
        CrystalInfo = smoothdataIngredients.crystalInfo,
        WeightWorkspace = weight_ws_name,
)
```
or import from mantid and call directly, or set it as a child algorithm, as before.

Reply:
Done.